### PR TITLE
[BUGFIX]: JavaScripts max int is 2^53 - 1, longs are bigger

### PR DESCRIPTION
**Pull request by fabianmenges**
Twitter uses very long IDs for entities. This causes issues because Java Script can not handle these long IDs and will **round** them by treating them as a float. 

This is a very hack fix, that will just convert every integer that is longer than the longest integer supported by JS to a string for both SQL Lab as well as the default table visualization. 
This works since both python and JS are dynamically typed, but its not great.

Any suggestions how we should handle this correctly?

### CODE REVIEW

**Comment by rumbin** [comment]
I think #3188 is related...

**Comment by rumbin** [comment]
Corresponding issue: #4005

**Comment by bolkedebruin** [comment]
I think using the BigInt package https://www.npmjs.com/package/big-integer might be smarter. Also DRY being applied would be nice and a test.

File: superset/viz.py [hunk 1]
```diff
@@ -935,7 +946,10 @@ def to_series(self, df, classed='', title_suffix=''):
             ys = series[name]
             if df[name].dtype.kind not in 'biufc':
                 continue
-            series_title = name
+            if isinstance(name, (list, tuple)):
```
Comment (xrmx) [inline superset/viz.py]: This hunk looks unrelated

File: superset/dataframe.py [hunk 2]
```diff
@@ -50,9 +52,17 @@ def size(self):
...
-        return [dict((k, _maybe_box_datetimelike(v))
-                     for k, v in zip(self.__df.columns, np.atleast_1d(row)))
+        data = [dict((k, _maybe_box_datetimelike(v))
+                for k, v in zip(self.__df.columns, np.atleast_1d(row)))
                 for row in self.__df.values]
+        for d in data:
+            for k, v in list(d.items()):
+                # if an int is too big for Java Script to handle
+                # convert it to a string
+                if isinstance(v, int):
```
Comment (xrmx) [inline superset/dataframe.py]: What about moving the constant and the loop in an helper?

**Review by xrmx — COMMENTED** [review]
Hopefully BigInt is coming in JS:
https://github.com/tc39/proposal-bigint

File: superset/viz.py (hunk 1 above)
Comment (fabianmenges) [reply superset/viz.py]: This is actually related. When you group by a numeric entity you can run into the same problem that the ID is to large to be displayed correctly in the legend or pop over. However we can just treat everything in the title a string since we can be sure its not used for anything mathematical.

File: superset/dataframe.py (hunk 2 above)
Comment (fabianmenges) [reply superset/dataframe.py]: Sure, will do.

**Comment by fabianmenges** [comment]
@bolkedebruin @xrmx I looked at the various BigInt packages in JS. The problem is that I don't think d3 will handle BigInts correctly, which is why I'm converting large numbers to string only in cases where I'm pretty sure we are not going to do any computations with them, specifically:

- The legend of line charts
- Any table visualization (SQL lab or table slice)

If a value is supposed to be displayed/ingested in a visualization (e.g. line char,  pie chart, ...) it is usually fine for it to loose precision since you usually don't care about the exact number anymore. 
Unless JS supports a real bigint and all of our visualization libraries support it we will probably have to convert big integers to string on a visualization by visualization level.

An alternative to what I did in this PR would be to define that all integers for the table visualizations are always converted to a string (regardless of their size) and then fix up the sorting logic for sorting on the client side.

File: superset/viz.py [hunk 3]
```diff
@@ -935,7 +946,10 @@ def to_series(self, df, classed='', title_suffix=''):
...
+                series_title = [str(title) for title in name]
```
Comment (mistercrunch) [inline superset/viz.py]: does that affect bar chart sorting?

**Review by mistercrunch — COMMENTED** [review]
This is a tough one to work around of... while your fix seems like it would work for dimensions, what happens with overflowing metrics?

File: superset/viz.py (hunk 3 above)
Comment (fabianmenges) [reply superset/viz.py]: It should not since we are treating the title as string in the JS anyways.

**Comment by fabianmenges** [comment]
@mistercrunch A couple screenshots how everything behaves:
**SQL Lab**: 
You can see that the bigint is rendered correctly when its a dimension, if you put a sum around it, it would still work
screen shot 2017-12-05 at 2 25 37 pm
**Table Viz**
Grouped by time and summed. The result shows a loss of precision: 
screen shot 2017-12-05 at 2 23 14 pm
Non Grouped the precision is maintained
screen shot 2017-12-05 at 2 24 31 pm

File: superset/dataframe.py (hunk 2 above)
Comment (mistercrunch) [reply superset/dataframe.py]: Good idea, `def handle_js_int_overflow(data)` or something

File: superset/viz.py [hunk 4]
```diff
@@ -421,11 +421,20 @@ def get_data(self, df):
             ):
                 del df[m]
 
-        return dict(
+        data = dict(
             records=df.to_dict(orient='records'),
             columns=list(df.columns),
         )
 
+        for d in data.get('records', dict()):
```
Comment (mistercrunch) [inline superset/viz.py]: let's also put this into a `BaseViz.handle_js_int_overflow` method

**Comment by jeffreythewang** [comment]
Any updates on this besides a rebase?
